zcash_primitives/src/transaction/components/orchard/burn_validation.rs

```diff
@@ -0,0 +1,118 @@
+use std::fmt;
+
+use orchard::note::AssetBase;
+
+use super::Amount;
+
+// FIXME: Consider making tuple (AssetBase, Amount) a new type.
+
+#[derive(Debug)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+pub enum BurnError {
+    DuplicateAsset,
+    NativeAsset,
+    ZeroAmount,
+}
+
+impl fmt::Display for BurnError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            BurnError::DuplicateAsset => write!(f, "Encountered a duplicate asset to burn."),
+            BurnError::NativeAsset => write!(f, "Cannot burn a native asset."),
+            BurnError::ZeroAmount => write!(f, "Cannot burn an asset with a zero amount."),
+        }
+    }
+}
+
+/// Validates burns for a bundle by ensuring each asset is unique, non-native, and has a non-zero value.
```

Comment:
Validates the burn values for a bundle by ensuring each asset is unique, non-native, and has a possitive value.

Reply:
Done.